File: Easy/MiddleOfLinkedList/MiddleOfLinkedList.cpp
// https://leetcode.com/problems/middle-of-the-linked-list/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* mid = head;
        int count = 1;
        while(head != nullptr){
            if(count%2==0){
                mid = mid->next;
            }
            count++;
            head = head->next;
        }


        return mid;
    }
};

File: Easy/MiddleOfLinkedList/MiddleOfLinkedList.py
# https://leetcode.com/problems/middle-of-the-linked-list/

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        mid = head
        updateMid = False
        while head:
            if updateMid:
                mid = mid.next
            updateMid=not updateMid
            head = head.next

        return mid